rse-bash-advanced.Rmd

```diff
@@ -702,6 +702,181 @@ $ find . -name "*.bak"
+here are the components of `PATH` listed one per line:
+
+```shell
+/Users/amira/bin
+/usr/local/git/bin
+/usr/bin
+/bin
+/usr/sbin
+/sbin
+/usr/local/bin
+```
+
+On our computer,
+there are actually three programs called `analyze`
+in three different directories:
+`/bin/analyze`,
+`/usr/local/bin/analyze`,
+and `/users/amira/analyze`.
+Since the shell searches the directories in the order they're listed in `PATH`,
+it finds `/bin/analyze` first and runs that.
+Notice that it will *never* find the program `/users/amira/analyze`
+unless we type in the full path to the program,
+since the directory `/users/amira` isn't in `PATH`.
+
+Let's show the value of the variable `HOME`:
+
+```shell
+$ echo HOME
+```
+
+```text
+HOME
+```
+
+That just prints "HOME", which isn't what we wanted
+(though it is what we actually asked for).
+Let's try this instead:
+
+```shell
+$ echo $HOME
+```
+
+```text
+/home/amira
+```
+
+The dollar sign tells the shell that we want the value of the variable
+rather than its name.
+This works just like wildcards:
+the shell does the replacement before running the program we've asked for.
+Thanks to this expansion, what we actually run is `echo /home/amira`,
+which displays the right thing.
+
+Creating a variable is easy—we just assign a value to a name using "=",
+and put quotes around the value if it contains spaces or special characters:
+
+```shell
+$ DEPARTMENT="Library Science"
+$ echo $DEPARTMENT
+```
+
+```text
+Library Science
+```
+
+To change the value, we just assign a new one:
+
+```shell
+$ DEPARTMENT="Information Science"
+$ echo $DEPARTMENT
+```
+
+```text
+Information Science
+```
+
+If we want to set some variables automatically every time we run a shell,
+we can put commands to do this in a file called `.bashrc` in our home directory.
+(The '.' character at the front prevents `ls` from listing this file
+unless we specifically ask it to using `-a`.
+The "rc" at the end is an abbreviation for "run control",
+which meant something really important decades ago,
+and is now just a convention everyone follows without understanding why.)
+For example,
+here are two lines in `/home/amira/.bashrc`:
+
+```text
+export DEPARTMENT="Library Science"
+export TEMP_DIR=/tmp
+export BACKUP_DIR=$TEMP_DIR/backup
+```
+
+These three lines create the variables `DEPARTMENT`,
+`TEMP_DIR`,
+and `BACKUP_DIR`,
+and export them so that any programs the shell runs can see them as well.
+Notice that `BACKUP_DIR`'s definition relies on the value of `TEMP_DIR`,
+so that if we change where we put temporary files,
+our backups will be relocated automatically.
+
+While we're here,
+it's also common to use the `alias` command to create shortcuts for things we frequently type.
+For example, we can define the alias `backup`
+to run `/bin/zback` with a specific set of arguments:
+
+```shell
+alias backup=/bin/zback -v --nostir -R 20000 $HOME $BACKUP_DIR
+```
+
+Aliases can save us a lot of typing, and hence a lot of typing mistakes.
```

Comment:
````suggestion
Aliases can save us a lot of typing, and hence a lot of typing mistakes.
The name of an alias can be the same as an existing command,
so we can use them to change the "default setting" a command runs with, e.g.

```shell
# Long list format including hidden files
alias ls=ls -la

# Print the file paths that were copied/moved
alias mv=mv -v
alias cp=cp -v

# Print the file paths that were removed
# and prompt if trying to remove move than three files
alias rm=rm -Iv
```
````

Reply:
Good - added.